From 69afaedf6462cd19689f743bc26e2ce7968a5f3d Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Wed, 25 Jul 2018 10:31:27 +0800
Subject: [PATCH 44/65] fix installed system boot from encrypt fs failed

1. While using cryptsetup to encrypt filesystem, it failed to boot
installed system.
-----------------
dracut: luksOpen /dev/sda3 luks-9cf2d2d9-849e-4ecb-a7f8-5918259306d6
Enter passphrase for /dev/sda3: Switched to clocksource tsc

device-mapper: table: 253:0: crypt: Error allocating crypto tfm
device-mapper: ioctl: error adding target to table
device-mapper: reload ioctl on  failed: No such file or directory
Failed to setup dm-crypt key mapping for device /dev/sda3.
Check that kernel supports aes-xts-plain64 cipher (check syslog for more info).
Wrong password
-----------------

The reason is the initramfs (generated by dracut) missed kernel module.

Config dracut to let initramfs has the kernel driver.

2. Since systemd in oe-core does not support cryptsetup which caused a
loop dependency with lvm2, so ommit systemd module in dracut.

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/payload/__init__.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/pyanaconda/payload/__init__.py b/pyanaconda/payload/__init__.py
index c3a74d5..bdeb7b8 100644
--- a/pyanaconda/payload/__init__.py
+++ b/pyanaconda/payload/__init__.py
@@ -855,6 +855,12 @@ class Payload(object):
         for subdir in ["log", "tmp"]:
             util.mkdirChain(util.getSysroot() + "/var/volatile/%s" % subdir)
 
+        with open(util.getSysroot() + "/etc/dracut.conf.d/cryptsetup.conf", "w") as f:
+            f.write('add_drivers+=" aesni_intel"\n')
+            f.write('add_drivers+=" aes_x86_64"\n')
+            f.write('omit_dracutmodules+=" systemd"\n')
+
+
         for kernel in self.kernelVersionList:
             log.info("recreating initrd for %s", kernel)
             if not flags.imageInstall:
-- 
2.7.4

